### ISSUE TITLE
Mouse jumps when changing monitors or when mouse is ungrabbed, if xinput Coordinate Transformation Matrix property was changed

### ISSUE THREAD

**Issue by fghsgh** [i3/i3#4169]
## I'm submitting a…

<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

When changing the focused monitor (e.g. with `$mod+2` with the default config), the mouse cursor first appears in the middle, but as soon as it is moved, it jumps to the edge of said monitor. The exception to this is the first time this happens (a.k.a. the first time you change monitors), in which case it works as expected. Furthermore, when this is done with a touchpad instead of a mouse, the mouse cursor jumps to the other side of the monitor. Each time you switch between touchpad and mouse, it doesn't happen once.

The following is a screen recording of this behavior: https://youtu.be/HppeOQ7pY_4
The logfile included below is from the same session. The only running application in the recording is a terminal emulator with ffmpeg to do the screen recording. I switch between mouse and touchpad several times throughout the video.

Furthermore, when a program (such as a fullscreen game) grabs the mouse and later ungrabs it, the mouse similarly jumps to the border of the window which, depending on the direction in which the mouse was moving, makes it lose focus.

## Expected Behavior

The mouse cursor doesn't jump but instead simply moves from the middle of the monitor as you move the mouse.

## Reproduction Instructions

No windows need to be open, and the default config can be used. This started happening while trying to turn off touchpad mouse acceleration. This included installing Synaptics, which made the regular mouse be handled by libinput but the touchpad by Synaptics. I suspect this may be related.

EDIT: As discussed in the post below, this happens if the xinput Coordinate Transformation Matrix is not set to the default. See the post below for more details.

## Environment

Output of `i3 --moreversion 2>&-`: (I temporarily upgraded to the latest development version but regularly use the latest release)
<pre>
Binary i3 version:  4.18-345-ge5992eed (2020-08-01, branch "next") © 2009 Michael Stapelberg and contributors
Running i3 version: 4.18.2 (2020-07-26) (pid 7489)
Loaded i3 config: /home/erik/.config/i3/config (Last modified: Fri 07 Aug 2020 03:47:01 PM CEST, 3986 seconds ago)

The i3 binary you just called: /home/erik/build/i3/x86_64-pc-linux-gnu/i3
The i3 binary you are running: i3
</pre>

This issue occurs even with the default config file. Turning off mouse warping fixes it when changing workspaces, but not when an application ungrabs the mouse and places it in the middle. Also, I would like to keep mouse warping enabled if possible.

<pre>
Logfile URL: https://logs.i3wm.org/logs/5724385621049344.bz2
</pre>

<pre>
- Linux Distribution & Version: Arch Linux x86_64, 5.7.12-arch1-1
- Are you using a compositor (e.g., xcompmgr or compton): no
</pre>

**Comment by fghsgh** [comment]
I have also discovered that clicking with the mouse on a split (which lets you resize the windows) also does something similar, but the specific location to which the mouse jumps depends on both the orientation of the split and on the location of the mouse itself.

**Comment by fghsgh** [comment]
I discovered that unplugging and replugging the mouse in question resolves the issue for the mouse, but not for the touchpad, as it can't be unplugged. In any case, this should be considered a workaround, not a fix, as the touchpad is still not working properly, and it requires manual effort with each session startup. Replugging the mouse more often because of this could result in damage to the USB port.

EDIT: After some more investigation, I discovered that this is caused by the "Coordinate Transformation Matrix" xinput property. I use this option to change the sensitivity of the mouse/touchpad, because xinput doesn't have a "speed" option, only an "accel" option. Thus, it seems i3 must be misinterpreting the transformation matrix somewhere. Replugging the mouse of course caused the settings to reset to the default `1 0 0 0 1 0 0 0 1`, which does not cause the issue. Furthermore, the exact coefficients used probably determine where the mouse will jump to, which explains why the touchpad and the mouse were jumping in opposite directions, as the mouse was set to `1 0 0 0 1 0 0 0 .75` and the touchpad to `1 0 0 0 1 0 0 0 2.5`.

I will now edit the title of this issue to reflect these new findings.

EDIT: I tried changing the translation components of the transformation matrix, as these don't affect actual mouse movement, but they affect this issue too. It was possible to get the mouse not to jump as much on a single monitor, but it seems impossible to correct it on all monitors simultaneously using exclusively the translation components. By changing the other elements of the matrix, it is possible to fix, but only by reverting to what is effectively still default behavior, like `2 0 0 0 2 0 0 0 2`, which completely disregards the initial reason for changing the coordinate transformation matrix in the first place.

EDIT: after inspecting the source code, it seems this is due to inproper usage of `xcb_warp_pointer()`. With this knowledge, I also found this issue to be a duplicate of #3598 and #4130, therefore I am closing this issue now.
